presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/ResourceGroupSpec.java

```diff
@@ -62,10 +65,16 @@ public ResourceGroupSpec(
             @JsonProperty("subGroups") Optional<List<ResourceGroupSpec>> subGroups,
             @JsonProperty("jmxExport") Optional<Boolean> jmxExport,
             @JsonProperty("softCpuLimit") Optional<Duration> softCpuLimit,
-            @JsonProperty("hardCpuLimit") Optional<Duration> hardCpuLimit)
+            @JsonProperty("hardCpuLimit") Optional<Duration> hardCpuLimit,
+            @JsonProperty("maxExecutionTime") Optional<Duration> maxExecutionTime,
+            @JsonProperty("maxTotalMemoryLimit") Optional<DataSize> maxTotalMemoryLimit,
+            @JsonProperty("maxCpuTime") Optional<Duration> maxCpuTime)
     {
         this.softCpuLimit = requireNonNull(softCpuLimit, "softCpuLimit is null");
         this.hardCpuLimit = requireNonNull(hardCpuLimit, "hardCpuLimit is null");
+        this.maxExecutionTime = maxExecutionTime;
+        this.maxTotalMemoryLimit = maxTotalMemoryLimit;
+        this.maxCpuTime = maxCpuTime;
```

Comment:
In constructors - we always check for not null values. `null`s are tricky to manage in our Java codebase - and hence we avoid nulls and put checkguards everywhere to ensure that there are no nulls.

```
this.perQueryExecutionTimeLimit = requireNonNull(perQueryExecutionTimeLimit, "perQueryExecutionTimeLimit is null");
this.perQueryTotalMemoryLimit = requireNonNull(perQueryTotalMemoryLimit, "perQueryTotalMemoryLimit is null");
this.perQueryCpuTimeLimit = requireNonNull(perQueryCpuTimeLimit, "perQueryCpuTimeLimit is null");
```